hack/generate-dnsmasq-site-policy-section.sh

```diff
@@ -63,20 +89,28 @@ EOF
 
 function generate_dnsmasq_body {
     cat <<EOF
-apiVersion: machineconfiguration.openshift.io/v1
-kind: MachineConfig
 metadata:
   labels:
     machineconfiguration.openshift.io/role: master
   name: 50-master-dnsmasq-configuration
+EOF
+    if [ "${ZTP_DEPLOY_WAVE}" -gt 0 ]; then
+        cat <<EOF
+  annotations:
+    ran.openshift.io/ztp-deploy-wave: "${ZTP_DEPLOY_WAVE}"
+EOF
+    fi
+    cat <<EOF
 spec:
   config:
+    ignition:
+      version: 3.1.0
```

Comment:
Why ignition version 3.1.0? I think we can use 3.4.0

Reply:
This is to align with the content generated here: https://github.com/openshift/assisted-service/blob/master/internal/network/manifests_generator.go#L134